kedro/extras/datasets/matplotlib/matplotlib_writer.py

```diff
@@ -146,7 +146,9 @@ def _save(
             and self._overwrite
             and self._fs.exists(self._filepath)
         ):
-            self._fs.rm(self._filepath, recursive=True)
+            self._fs.rm(
+                get_filepath_str(self._filepath, self._protocol), recursive=True
```

Comment:
Following my comment in #1085 I think what we're interested in here (in both lines 147 and 150) would be `get_filepath_str(save_path, self._protocol)`. Does that make sense to you also?